File: benchmarks/java/integer_boolean/Java_Primitive_Integer/LiftExec.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class LiftExec {



    private Lift _Lift = new Lift();



    private BInteger counter;

    public LiftExec() {
        counter = new BInteger(0);
    }

    public void simulate() {
        while((counter.less(new BInteger(3000))).booleanValue()) {
            BInteger i = null;
            i = new BInteger(0);
            while((i.less(new BInteger(100))).booleanValue()) {
                this._Lift.inc();
                i = i.plus(new BInteger(1));
            }
            BInteger _i = null;
            _i = new BInteger(0);
            while((_i.less(new BInteger(100))).booleanValue()) {
                this._Lift.dec();
                _i = _i.plus(new BInteger(1));
            }
            counter = counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        LiftExec exec = new LiftExec();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}

File: benchmarks/C++_Primitives/LiftExec.cpp
#include <iostream>
#include <string>
#include "BUtils.hpp"
#include "BInteger.hpp"
#include "BBoolean.hpp"
#include "BTuple.hpp"
#include "Lift.cpp"

#ifndef LiftExec_H
#define LiftExec_H

using namespace std;

class LiftExec {

    public:

    private:


        Lift _Lift;


        BInteger counter;

        BTuple<BInteger, BInteger> tuple;

    public:

        LiftExec() {
            counter = (BInteger(0));
            tuple = (BTuple<BInteger, BInteger>(counter, counter));
        }

        void simulate() {
            while((counter.less((BInteger(10000000)))).booleanValue()) {
                BInteger i;
                i = (BInteger(0));
                while((i.less((BInteger(100)))).booleanValue()) {
                    this->_Lift.inc();
                    i = i.plus((BInteger(1)));
                }
                BInteger _i;
                _i = (BInteger(0));
                while((_i.less((BInteger(100)))).booleanValue()) {
                    this->_Lift.dec();
                    _i = _i.plus((BInteger(1)));
                }
                counter = counter.plus((BInteger(1)));
            }
        }
};

int main() {
    LiftExec exec;
    exec.simulate();
    return 0;
}

#endif